ami/ml/schemas.py

```diff
@@ -62,6 +62,65 @@ class PipelineRequest(pydantic.BaseModel):
 class PipelineResponse(pydantic.BaseModel):
     # pipeline: PipelineChoice
     pipeline: str
-    total_time: float
+    total_time: float | None
     source_images: list[SourceImageResponse]
     detections: list[DetectionResponse]
+    errors: list | str | None = None
+
+
+class PipelineStageParam(pydantic.BaseModel):
+    """A configurable parameter of a stage of a pipeline."""
+
+    name: str
+    key: str
+    category: str = "default"
+
+
+class PipelineStage(pydantic.BaseModel):
+    """A configurable stage of a pipeline."""
+
+    key: str
+    name: str
+    params: list[PipelineStageParam] = []
+    description: str | None = None
+
+
+class ProjectConfig(pydantic.BaseModel):
+    name: str
+
+
+class AlgorithmConfig(pydantic.BaseModel):
+    name: str
+    key: str
+
+
+class PipelineConfig(pydantic.BaseModel):
+    """A configurable pipeline."""
+
+    name: str
+    slug: str
+    version: int
+    description: str | None = None
+    algorithms: list[AlgorithmConfig] = []
+    stages: list[PipelineStage] = []
+    projects: list[ProjectConfig] = []
+
+
+class BackendStatusResponse(pydantic.BaseModel):
+    timestamp: datetime.datetime
+    request_successful: bool
+    pipeline_configs: list[PipelineConfig] = []
+    error: str | None = None
+    server_live: bool
+    pipelines_online: list[str] | str
+    endpoint_url: str
+
+
+class PipelineRegistrationResponse(pydantic.BaseModel):
```

Comment:
I don't think you need this PipelineRegistrationResponse. Registration is done on the Antenna side, so that's handled by the Django Rest Framework serlializers & viewsets

Reply:
I currently use PipelineRegistrationResponse in the viewset `ami-platform/ami/ml/models/backend.py`. This is similar to how the get_status view returns a BackendStatusResponse. Should I remove both of these response schemas?

Reply:
Ah yes, okay that makes sense! Thank you keep it